xray/core/dataarray.py

```diff
@@ -912,6 +918,22 @@ def to_series(self):
         index = self.coords.to_index()
         return pd.Series(self.values.reshape(-1), index=index, name=self.name)
 
+    def to_masked_array(self, copy=True):
```

Comment:
I don't see much value in the mask option or letting the user choose not to mask out nulls. If you want that, you can just use the MaskedArray constructor directly.